tests/ui/excessive_precision.rs

```diff
@@ -78,4 +78,7 @@ fn main() {
     const NEG_INF1: f32 = -1.0e+33f32;
     const NEG_INF2: f64 = -1.0e+3300f64;
     const NEG_INF3: f32 = -3.40282357e+38_f32;
+
+    // issue #12954
+    const _: f64 = 3.0000000000000000e+00;
```

Comment:
While we're here, can you also add `3.0000000000000000` as a test case?

It currently doesn't emit a warning (because it takes a different code path in the lint code), which is fine, but just to make sure it doesn't regress later, because the fix in this PR wouldn't apply to that literal if it got into the numeric literal code as there's no exponent and would end up with "3" as the suggestion again.

Reply:
Sure